src/Api/AdminConsole/Controllers/BaseAdminConsoleController.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Bit.Core.AdminConsole.Utilities.v2;
+using Bit.Core.AdminConsole.Utilities.v2.Results;
+using Bit.Core.Models.Api;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Bit.Api.AdminConsole.Controllers;
+
+public abstract class BaseAdminConsoleController : Controller
+{
+    protected static IResult Handle<T>(CommandResult<T> commandResult) =>
+        commandResult.Match<IResult>(
+            error => error switch
+            {
+                BadRequestError badRequest => TypedResults.BadRequest(new ErrorResponseModel(badRequest.Message)),
+                NotFoundError notFound => TypedResults.NotFound(new ErrorResponseModel(notFound.Message)),
+                InternalError internalError => TypedResults.Json(
+                    new ErrorResponseModel(internalError.Message),
+                    statusCode: StatusCodes.Status500InternalServerError),
+                _ => TypedResults.Json(
+                    new ErrorResponseModel(commandResult.AsError.Message),
+                    statusCode: StatusCodes.Status500InternalServerError
+                )
+            },
+            _ => TypedResults.NoContent()
+        );
+
+    protected static IResult Handle(BulkCommandResult commandResult) =>
+        commandResult.Result.Match<IResult>(
+           error => error switch
+           {
+               NotFoundError notFoundError => TypedResults.NotFound(new ErrorResponseModel(notFoundError.Message)),
+               _ => TypedResults.BadRequest(new ErrorResponseModel(error.Message))
+           },
+           _ => TypedResults.NoContent()
+        );
```

Comment:
This is not generally how we'd handle a `BulkCommandResponse`; see the `BulkDeleteAccount` endpoint as a better example. (i.e. it's usually mapped to a BulkResponseModel with an OK response because there is no single success or failure.)

I see that you're currently using this on the (non-bulk) `DeleteAccount` endpoint which is why it "works", but I think that indicates that the non-bulk endpoint shouldn't be handling a `BulkCommandResponse`.